docs/operator-manual/applicationset/Appset-Any-Namespace.md

```diff
@@ -23,6 +23,38 @@ This feature needs [App in any namespace](../app-any-namespace.md) feature activ
 
 This feature can only be enabled and used when your Argo CD ApplicationSet controller is installed as a cluster-wide instance, so it has permissions to list and manipulate resources on a cluster scope. It will *not* work with an Argo CD installed in namespace-scoped mode.
 
+### Scm Providers secrets consideration
+
+By allowing ApplicationSet in any namespace you must be aware that any secrets can be exfiltrated using scmProvider or pullRequest generators.
+
+Here is an example:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: myapps
+spec:
+  generators:
+  - scmProvider:
+      gitea:
+        # The Gitea owner to scan.
+        owner: myorg
+        # The Gitea instance url
+        api: https://gitea.mydomain.com/
+        # If true, scan every branch of every repository. If false, scan only the default branch. Defaults to false.
+        allBranches: true
+        # Reference to a Secret containing an access token. (optional)
+        tokenRef:
+          secretName: gitea-token
+          key: token
+  template:
+```
+
+By setting `spec.generators[n].scmProvider.gitea.api=http://my-service.my-namespace.svc.cluster.local` user can send all request to a Pod that will log incoming requests including headers with tokens. By changing the `tokenRef` user can read in clear all secrets.
+
+Therefore administrator must restrict the urls of the allowed SCM Providers (example: `https://git.mydomain.com/,https://gitlab.mydomain.com/`) by setting the environment variable `ARGOCD_APPLICATIONSET_CONTROLLER_ALLOWED_SCM_PROVIDERS` to argocd-cmd-params-cm `applicationsetcontroller.allowed.scm.providers`. If another url is used, it will be rejected by the applicationset controller.
+
```

Comment:
````suggestion

For example:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-cmd-params-cm
data:
  applicationsetcontroller.allowed.scm.providers: https://git.mydomain.com/,https://gitlab.mydomain.com/
```
````

Again, are protocols allowed here? Or just the domain names?

Reply:
According to me it would be simpler to ask people to put the protocol. However if you prefer I can change the method `ScmProviderAllowed` to support domain name format instead.

Reply:
Nah, including the protocol works for me! Just as long as that's what we expect in the `api` fields.